services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV2Spec.groovy

```diff
@@ -1193,6 +1193,48 @@ class FlowCrudV2Spec extends HealthCheckSpecification {
         }
     }
 
+    @Ignore("https://github.com/telstra/open-kilda/issues/3077")
+    @Tidy
+    def "Flow is healthy when assigned transit vlan matches the flow endpoint vlan"() {
+        given: "We know what the next transit vlan will be"
+        //Transit vlans are simply picked incrementally, so create a flow to see what is the current transit vlan
+        //and assume that the next will be 'current + 1'
+        def helperFlow = flowHelperV2.randomFlow(topologyHelper.notNeighboringSwitchPair)
+        flowHelperV2.addFlow(helperFlow)
+        def helperFlowInfo = database.getFlow(helperFlow.flowId)
+        def nextTransitVlan = database.getTransitVlan(helperFlowInfo.forwardPath.pathId).get().vlan + 1
+
+        when: "Create flow with endpoint vlans matching the next transit vlan"
+        def flow = flowHelperV2.randomFlow(topologyHelper.notNeighboringSwitchPair).tap {
+            it.source.vlanId = nextTransitVlan
+            it.destination.vlanId = nextTransitVlan
+        }
+        flowHelperV2.addFlow(flow)
+        //verify that our 'nextTransitVlan' assumption was correct
+        assert database.getTransitVlan(database.getFlow(flow.flowId).forwardPath.pathId).get().vlan == nextTransitVlan
+
+        then: "Flow is pingable"
+        with(northbound.pingFlow(flow.flowId, new PingInput())) {
+            it.forward.pingSuccess
+            it.reverse.pingSuccess
+        }
+
+        and: "Flow passes flow validation"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+
+        and: "Involved switches pass switch validation"
+        pathHelper.getInvolvedSwitches(flow.flowId).each {
+            with(northbound.validateSwitch(it.dpId)) { validation ->
+                validation.verifyRuleSectionsAreEmpty(["missing", "misconfigured", "excess"])
+                validation.verifyMeterSectionsAreEmpty(["missing", "misconfigured", "excess"])
+            }
+        }
+
```

Comment:
Do you mind if we check that system allows to pass traffic through the flows?

Reply:
yeah, had the same thoughts. I'll remove ping then and add traffic check. Because pings don't really leverage the ingress rule itself